astrowidgets/core.py

```diff
@@ -237,6 +237,24 @@ def _mouse_click_cb(self, viewer, event, data_x, data_y):
 #         from IPython.display import display
 #         return display(self._widget)
 
+    def load_file(self, filename, **kwargs):
+        """
+        Load a given file into the viewer. This relies on the viewer
+        to identify the best file handler to use.
+
+        Parameters
+        ----------
+        filename : str
+            File name.
+
+        kwargs : dict
+            Extra keywords to be passed into the viewer's file handler.
+
+        """
```

Comment:
@pllim, you might want to use:
```python
from ginga.util import loader
image = loader.load_data(filename, **kwargs)
if not isinstance(image, AstroImage):
    raise ValueError("Default index in '{}' does not reference an image".format(filename))
self._viewer.set_image(image)
```

Reply:
Do you have a parameter for specifying HDU or some kind of index in the astrowidget load API?  Because you could pass it in to `idx` in this `load_data` call.

Reply:
This new method is meant more for ASDF than for FITS, though ideally it would work on either one if the Ginga file loader registry is set up properly.